src/pudl/transform/ferc1.py

```diff
@@ -717,63 +720,75 @@ def drop_null_data_rows(self, df: pd.DataFrame) -> pd.DataFrame:
-            )
-            .pipe(self.simplify_strings_for_table)
-            .pipe(self.clean_strings_for_table)
-            .pipe(self.oob_to_nan_for_table)
-            .pipe(self.convert_units_in_table)
-            .pipe(self.convert_table_dtypes)
-            .pipe(plants_steam_assign_plant_ids, ferc1_fuel_df=transformed_fuel)
-        )
-        plants_steam_validate_ids(plants_steam_combo)
-        return plants_steam_combo
-
-    def process_dbf(self, raw_dbf):
-        """Modifications of the dbf plants_steam_ferc1 table before concat w/ xbrl."""
-        plants_steam_dbf = self.rename_columns(
-            raw_table=raw_dbf,
-            source="dbf",
-        ).pipe(self.assign_record_id, source_ferc1=Ferc1Source.DBF)
-        return plants_steam_dbf
-
-    def process_xbrl(self, raw_xbrl_instant, raw_xbrl_duration):
-        """Modifications of the xbrl plants_steam_ferc1 table before concat w/ xbrl."""
-        plants_steam_xbrl = (
-            self.merge_instant_and_duration_tables(
-                duration=raw_xbrl_duration,
-                instant=raw_xbrl_instant,
+        self.plants_steam = (
+            self.concat_dbf_xbrl(
+                raw_dbf=raw_dbf,
+                raw_xbrl_instant=raw_xbrl_instant,
+                raw_xbrl_duration=raw_xbrl_duration,
             )
+            .pipe(self.normalize_strings_multicol, params=self.params.normalize_strings)
+            .pipe(self.nullify_outliers_multicol, params=self.params.nullify_outliers)
             .pipe(
-                self.rename_columns,
-                source="xbrl",
+                self.categorize_strings_multicol, params=self.params.categorize_strings
             )
-            .pipe(self.add_report_year_column)
-            .pipe(
-                self.assign_record_id,
-                source_ferc1=Ferc1Source.XBRL,
+            .pipe(self.convert_units_multicol, params=self.params.convert_units)
+            .pipe(self.drop_null_data_rows)
+        )
+        self.plants_steam = plants_steam_assign_plant_ids(
+            self.plants_steam,
+            ferc1_fuel_df=transformed_fuel,
+            fuel_categories=list(
+                FuelFerc1TableTransformer()
+                .params.categorize_strings["fuel_type_code_pudl"]
+                .categories.keys()
+            ),
+        ).pipe(self.enforce_schema)
+        plants_steam_validate_ids(self.plants_steam)
+        return self.plants_steam
+
+    def drop_null_data_rows(self, df):
+        """Drop rows with completely null or zero values."""
+        pre_drop_len = len(df)
+        non_data_cols = [
+            "record_id",
+            "utility_id_ferc1",
+            "plant_name_ferc1",
+            "report_year",
+            "entity_id",
+            "date",
+            "start_date",
+            "end_date",
+            "OrderNumber",
+            "PlantName",
+        ]
+        cols_to_check_nulls = [col for col in df if col not in non_data_cols]
```

Comment:
I don't understand the logic in this function. Can you either explain what's happening more fully in the docstring, or re-write the code to be more narrative?

You've got columns to check for nulls, but it's only the columns that are **not** in the columns that are **not** data, and then you've got a negation operation in the selector mask and the mask is doing like three things at once.

Reply:
this is very fair! I didn't document this method properly! I'll add info.

Reply:
this got moved/generalized and i added much more documentation